### ISSUE TITLE
Wine Support gone? 

### ISSUE THREAD

**Issue by EctoOne** [andresdelcampo/GameList_Editor#5]
Hi, I was looking for a GameList editor that works directly on a Pi(4) and wasn't really able to find one. But I remembered that this program was compatible with Wine and tried it on TwisterOS with the whole Box86/Wine thing, but sadly the latest version from here didn't really work. It started but I wasn't able to open the file browser to pick a rom folder without getting an error. I then tried the version from Neeeeb and it worked. So I guess somewhere on the way there was something added/changed that broke Wine Support (at least on the Pi).

Since I only needed it to have a better way to edit some descriptions, using the old version is fine for me but maybe it would be possible to bring Wine Support back.

**Comment by andresdelcampo** [comment]
I understand. I wonder what part broke it -opening a file browser should not have changed much, maybe due to the Delphi upgrade. I am not sure how easy or hard this would be without potentially dropping something else -like video support, but I will keep it in mind to see if I can do something about it. What kind of error were you seeing? That might make troubleshooting easier when getting to it. Thanks!

**Comment by EctoOne** [comment]
Errors when starting:

1.

```
Property cannot be translated.
Add
TP_GlobalIgnoreClassProperty(TWindowsMediaPlayer,'URL')
to your source code or use
TP_Ignore (self,'URL') to prevent this message.
Reason: OLE error 80004001.
```

2.

```
Retranslate was called on an object that has not
been translated before. Please use
TranslateComponent() before
RetranslateComponent().
```

Error when opening file browser:

```
Access violation at address 0085C39F in module
'GameListEditor.exe'. Read of address 00000008.
```


I also tried some older releases and the latest working version is GameList_Editor_2021_12_1.zip. The one before video support was added. Was that also the version where you had to use a different version of Delphi? Because I noticed that the window decorations have changed, but I don't remember in what version that was. Because I noticed another OLE error when I tried to close the window with the X button. I had to close it by using the menu instead.

**Comment by andresdelcampo** [comment]
Thanks for the info, your investigation will be very valuable. Given your interest, even if it is just one person asking, I will dedicate time to see if I can fix as much as possible in the following weeks. I never tried Wine, but this is a good reason to start.

Error 1 is kind of expected... the video support is implemented via Windows Media Player ActiveX and its associated codecs (very Windows dependent indeed, but anything from Windows 7 was compatible -I guess Wine is not :( ) -so that is a precondition for it. I could investigate if there is a way of starting up with video support disabled if that is stopping scenarios and the video is not a must. I am not sure I could provide video to non-Windows -or not pure Windows- scenarios.

I have no idea yet on error 2. That I would need to investigate. Sounds like Delphi internals handling localization, but I have not checked the sources yet. And the last one will need some investigation on the changes. It is strange that something as simple as browsing for a file could cause an issue, sounds like latest Delphi but I could investigate a bit. 

Yes, the first v2022 is a HUGE change compared to last year's releases where some of the components had been broken for years -like scraping- and I brought those back to life, including Delphi upgrade, components for modern web requests as the former were not working with the current server TLS requirements, etc, etc.  Please check the release description for more details. 

Bottom line, stay tuned, I will have a look short term and let you know more -hopefully fix some :)

**Comment by EctoOne** [comment]
Thanks, the support is very appreciated but you can take your time. As I said, I only needed it easily copy/paste some descriptions for some rom hacks. And since Skraper works fine on Linux/Pi with Mono, I don't know if there is really that much of interest in a GameList editor that works on Linux and has a huge amount of features. Also I personally don't care about videos, so I'm fine with what I have. Although I kind of miss an option to change the marquee and not just the image. Because that is actually something I do use. But again, Skraper does its job and replacing a wrong image doesn't really need a GUI.

**Comment by vladmix77** [comment]
I have the same problem. With the launch in wine.

**Comment by andresdelcampo** [comment]
Ok, thanks for the feedback. I will have a look in the next few weeks at the wine use case to see if I can fix it -or at least prevent it from crashing by disabling the videos.

**Comment by andresdelcampo** [comment]
I have not forgotten about this... 
I have a repro in an Ubuntu 24.04 VM with WineHQ:

![image](https://github.com/user-attachments/assets/b763081d-90d2-4a37-ad58-fb91b66e4bcd)

**Comment by andresdelcampo** [comment]
Good news. This executable should bring Wine support back by disabling Windows Media Player. 

I had to go through a new VM creation after losing my Virtualbox 6 Linux VM not compatible with VB 7 in Windows 11... but after some hassle I got one ready and installed Wine for the first time. A few hours, hopefully worth the time to get the new GameList Editor also available in other OSs. For the time being Windows Media Player is disabled in Wine though. It would require a full replacement also for Windows to a different video player library. 

Let me know what you think:
https://github.com/andresdelcampo/GameList_Editor/releases/tag/2023.12.2

**Comment by andresdelcampo** [comment]
Ok, just let me know if there is anything else. Closing.
